tensorflow/core/lib/random/random_distributions.h

```diff
@@ -95,47 +133,57 @@ template <class Generator>
 class UniformDistribution<Generator, bfloat16> {
  public:
   // The number of elements that will be returned.
-  static constexpr int kResultElementCount = Generator::kResultElementCount;
+  // Set the number to be greater equal to Eigen packet size of type,
+  // so computations can be vectorized using SIMD.
+  static constexpr int kVectorLength =
+      Eigen::internal::packet_traits<bfloat16>::size;
+  static constexpr int kResultElementCount =
+      (kVectorLength > Generator::kResultElementCount)
+          ? kVectorLength
+          : Generator::kResultElementCount;
   // Cost of generation of a single element (in cycles).
   static constexpr int kElementCost = 3;
   // Indicate that this distribution may take variable number of samples
   // during the runtime.
   static constexpr bool kVariableSamplesPerOutput = false;
   typedef Array<bfloat16, kResultElementCount> ResultType;
   typedef bfloat16 ResultElementType;
+  // Helper definiation for the format function.
```

Comment:
Nit: typo
```suggestion
  // Helper definition for the format function.
```
Please also fix this in the class for `float` as well.

Reply:
Done.